src/EnergyPlus/DXCoils.cc

```diff
@@ -2209,12 +2209,12 @@ namespace DXCoils {
 
 			// Only required for reverse cycle heat pumps
 			DXCoil( DXCoilNum ).DefrostEIRFT = GetCurveIndex( Alphas( 10 ) ); // convert curve name to number
-			if ( SameString( Alphas( 11 ), "ReverseCycle" ) && SameString( Alphas( 12 ), "OnDemand" ) ) {
+			if ( SameString( Alphas( 11 ), "ReverseCycle" ) ) {
```

Comment:
I remember adding this way back when I thought the curve was only used for OnDemand control type. The multispeed DX heating coil only checks for ReverseCycle so no need to correct that object's GetInput.